src/core/thread/mle_router.cpp

```diff
@@ -1427,7 +1427,8 @@ void MleRouter::UpdateRoutes(const RouteTlv &aRoute, uint8_t aRouterId)
     uint8_t cost;
     uint8_t linkQuality;
     bool    update;
-    bool    changed = false;
+    bool    requireResetAdv = false;
```

Comment:
nit: how about rename to something like `resetAdvInterval` (we are resetting the interval, not the advertisement itself).